tools/cosmovisor/process.go

```diff
@@ -168,6 +172,62 @@ func (l Launcher) doBackup() error {
 	return nil
 }
 
+func (l Launcher) doCustomPreUpgrade() error {
+	if l.cfg.CustomPreupgrade == "" {
+		return nil
+	}
+
+	// check if upgrade-info.json is not empty.
+	var uInfo upgradetypes.Plan
+	upgradeInfoFile, err := os.ReadFile(l.cfg.UpgradeInfoFilePath())
+	if err != nil {
+		return fmt.Errorf("error while reading upgrade-info.json: %w", err)
+	}
+
+	if err = json.Unmarshal(upgradeInfoFile, &uInfo); err != nil {
+		return err
+	}
+
+	if uInfo.Name == "" {
```

Comment:
nit: we could use the validate function on the plan

Reply:
changed code to use ValidateBasic

Reply:
should the test be removed completely since UpgradeBinary already checks?